test/test-utils.js

```diff
@@ -90,24 +90,38 @@ function ensureVirtualenv(venvName) {
   var venvDir = path.join(venvsBaseDir, venvName);
   try {
     fs.accessSync(venvDir, fs.R_OK);
+    return false;
   } catch (e) {
-    var revert = function () {};
-    if (process.env.VIRTUAL_ENV) {
-      revert = deactivateVirtualenv();
+    createVenv(venvDir);
+    return true;
+  }
+}
+
+function createVenv(venvDir) {
+  var revert = function () {};
+  if (process.env.VIRTUAL_ENV) {
+    revert = deactivateVirtualenv();
+  }
+  try {
+    var proc = subProcess.executeSync('virtualenv', [venvDir]);
+    if (proc.status !== 0) {
+      console.error(proc.stdout.toString() + '\n' + proc.stderr.toString());
+      throw new Error('Failed to create virtualenv in ' + venvDir);
     }
-    try {
-      var proc = subProcess.executeSync('virtualenv', [venvDir]);
+    if (process.env.PIP_VER) {
+      proc = subProcess.executeSync(
+        path.resolve(venvDir, binDirName, 'python'),
+        ['-m', 'pip', 'install', `pip==${process.env.PIP_VER}`]
```

Comment:
does this install the desired pip inside the virtualenv?

Reply:
Yes, it is calling the virtualenv's python.

(Have to use `python -m pip install pip` to upgrade pip.)